waspc/src/Wasp/Generator/DbGenerator/Jobs.hs

```diff
@@ -46,31 +47,38 @@ migrateDev projectDir migrateArgs =
     scriptArgs =
       if System.Info.os == "darwin"
         then -- NOTE(martin): On MacOS, command that `script` should execute is treated as multiple arguments.
-          ["-Fq", "/dev/null"] ++ prismaMigrateCmd
+          ["-Fq", "/dev/null"] ++ Shell.getShellArgValues prismaMigrateCmd
         else -- NOTE(martin): On Linux, command that `script` should execute is treated as one argument.
-          ["-feqc", unwords prismaMigrateCmd, "/dev/null"]
+        -- NOTE(miho): Since we are passing the arguments as a single string, we need to wrap the arguments
+        --  which might contain spaces in them. We are using Shell.Quoted to know which
+        --   arguments should be wrapped in quotes and which shouldn't.
+        --   Specifically, we are using `showShellArgs` to wrap file paths in quotes. Other arguments
+        --   are not wrapped in quotes since we know their values and they don't contain spaces.
+          ["-feqc", Shell.showShellArgs prismaMigrateCmd, "/dev/null"]
 
     -- NOTE(martin): For this to work on Mac, filepath in the list below must be as it is now - not
     -- wrapped in any quotes.
     -- NOTE(martin): We do "--skip-seed" here because I just think seeding happening automatically
     --   in some situations is too aggressive / confusing.
     prismaMigrateCmd =
-      [ absPrismaExecutableFp projectDir,
+      [ Shell.Quoted $ absPrismaExecutableFp projectDir,
         "migrate",
         "dev",
         "--schema",
-        SP.fromAbsFile schemaFile,
+        Shell.Quoted $ SP.fromAbsFile schemaFile,
         "--skip-generate",
         "--skip-seed"
       ]
```

Comment:
So these are all ShellCommandArg types, but the one that are string literals are being cast into Raw due to extension OverloadedStringLiterals + ShellCommandArg being isString?